doc/Language/iterating.pod6

```diff
@@ -4,136 +4,321 @@
 
 =SUBTITLE Functionalities available for visiting all items in a complex data structure
 
-=head1 The C<Iterator> and C<Iterable> roles
+Similar to other many programming languages, Raku makes a fine distinction among
+the terms I<iteration>, I<iterable>, and I<iterator>. Familiarizing yourself
+with them is key for understanding the Raku constructs that implement these
+concepts:
 
-Raku is a functional language, but functions need something to hold on to when
-working on complex data structures. In particular, they need a uniform interface
-that can be applied to all data structures in the same way. One of these kind of
-interfaces is provided by the L<Iterator|/type/Iterator> and
-L<Iterable|/type/Iterable> roles.
+=item An B<iteration> is a mechanism by which a block of code is executed repeatedly.
 
-The C<Iterable> role is relatively simple. It provides a stub for the
-C<iterator> method, which is the one actually used by statements such as C<for>.
-C<for> will call C<.iterator> on the variable it precedes, and then run a block
-once for every item. Other methods, such as array assignment, will make the
-C<Iterable> class behave in the same way.
+=item An B<iterable> is an object that produce an iterator.
+
+=item An B<iterator> is an object that performs an iteration. In other words, an
+iterator knows how to compute the next value during an iteration.
+
+Thus, an B<iterator> contains the state that moves as it works through the
```

Comment:
"moves" or "changes"? Does it "hold" the value of the state or "contains" the state? Arguably, the state would be contained in iterable.